sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/BlockBlobInputOutputStreamTest.groovy

```diff
@@ -173,4 +174,149 @@ class BlockBlobInputOutputStreamTest extends APISpec {
+        inputStream.read()
+
+        // Modify the blob again.
+        outStream = bc.getBlobOutputStream(true)
+        outStream.write(randomBytes, 1 * Constants.MB, 5 * Constants.MB)
+        outStream.close()
+
+        when:
+        inputStream.read()
+
+        then: "Exception should not be thrown even though blob was modified"
+        notThrown(IOException)
+    }
+
+    @Requires({ liveMode() })
+    def "Input stream concurrency control none error"() {
+        setup:
+        int length = 6 * Constants.MB
+        byte[] randomBytes = getRandomByteArray(length)
+        def blobContainerClient = versionedBlobServiceClient.createBlobContainer(containerName)
+        def blobClient = blobContainerClient.getBlobClient(generateBlobName())
+        BlobOutputStream outStream = blobClient.getBlockBlobClient().getBlobOutputStream()
+        outStream.write(randomBytes, 1 * Constants.MB, 5 * Constants.MB)
+        outStream.close()
+        def properties = blobClient.getProperties()
+
+        when:
+        blobClient.openInputStream(new BlobInputStreamOptions().setBlockSize(1).setConsistentReadControl(ConsistentReadControl.NONE)
+            .setRequestConditions(new BlobRequestConditions().setIfMatch(properties.getETag())))
+
+        then:
+        thrown(UnsupportedOperationException)
+
+        when:
+        blobClient.getVersionClient(properties.getVersionId()).openInputStream(new BlobInputStreamOptions().setBlockSize(1).setConsistentReadControl(ConsistentReadControl.NONE))
+
+        then:
+        thrown(UnsupportedOperationException)
+    }
+
+    @Requires({ liveMode() })
+    def "Input stream concurrency control etag"() {
+        setup:
+        int length = 6 * Constants.MB
+        byte[] randomBytes = getRandomByteArray(length)
+        def blobContainerClient = versionedBlobServiceClient.createBlobContainer(containerName)
+        def blobClient = blobContainerClient.getBlobClient(generateBlobName())
+        BlobOutputStream outStream = blobClient.getBlockBlobClient().getBlobOutputStream()
+        outStream.write(randomBytes, 1 * Constants.MB, 5 * Constants.MB)
+        outStream.close()
+        def properties = blobClient.getProperties()
+
+        when: "Use recent eTag"
+        def inputStream = blobClient.openInputStream(new BlobInputStreamOptions().setConsistentReadControl(ConsistentReadControl.ETAG)
+            .setRequestConditions(new BlobRequestConditions().setIfMatch(properties.getETag())))
+
+        then:
+        def outputStream = new ByteArrayOutputStream()
+        def b
+        try {
+            while ((b = inputStream.read()) != -1) {
+                outputStream.write(b)
+            }
+        } catch (IOException ex) {
+            throw new UncheckedIOException(ex)
+        }
+        byte[] randomBytes2 = outputStream.toByteArray()
+        assert randomBytes2 == Arrays.copyOfRange(randomBytes, 1 * Constants.MB, 6 * Constants.MB)
+
```

Comment:
I think we should also have a case where readControl.ETAG is set, but no request conditions are passed to ensure we will pick the latest etag if none is specified.

Reply:
And some validation that etag and version can't both be set for both the etag and version options.

Reply:
I think that is what "Input stream etag lock default" tests for. I can add tests for the etag and version set at same time ones

Reply:
Added the other tests too